### ISSUE TITLE
Collection sources  seems to break `getB` for a 2d array

### ISSUE THREAD

**Issue by apdioguardi** [magpylib/magpylib#507]
**Describe the bug**
Collection of current sources (in this case two loops) seems to break getB for a 2d (or higher) array.

**To Reproduce**

```python
import numpy as np
import magpylib as magpy
import scipy
from scipy.spatial.transform import Rotation

print("numpy version =", np.__version__)
print("magpylib version =", magpy.__version__)
print("scipy version =", scipy.__version__)

# create magpylib loop current sources

# rotation objects
r_x_90 = Rotation.from_rotvec(np.pi/2*np.array([1, 0, 0]))
r_y_90 = Rotation.from_rotvec(np.pi/2*np.array([0, 1, 0]))

# loop 1
loop_1 = magpy.current.Loop(current=100,
                            diameter=20,
                            position=(1,20,10),
                            orientation=r_x_90)
# loop 2
loop_2 = magpy.current.Loop(current=2,
                            diameter=40,
                            position=(20,10,1),
                            orientation=r_y_90)
# collection
loop_collection = magpy.Collection(loop_1, loop_2)

# display the spatial configuration of shims
#loop_collection.show()#backend='plotly')

# single observer position tests
observer_position = np.array([0,0,0])

superposed_B = magpy.getB(loop_1, observer_position) + magpy.getB(loop_2, observer_position)

collection_B = magpy.getB(loop_collection, observer_position)

print("Single observer at position =", observer_position)

print("superposed_B =")
print(superposed_B)

print("collection_B =")
print(collection_B)

print("allclose =")
print(np.allclose(superposed_B, collection_B))

observer_position = np.array([1,1,1])

superposed_B = magpy.getB(loop_1, observer_position) + magpy.getB(loop_2, observer_position)

collection_B = magpy.getB(loop_collection, observer_position)

print("Single observer at position =", observer_position)

print("superposed_B =")
print(superposed_B)

print("collection_B =")
print(collection_B)

print("allclose =")
print(np.allclose(superposed_B, collection_B))

# single observers work as expected

# multiple observer positions at once in a (2,3) shaped np array

observer_positions = np.array([[0,0,0], [1,1,1]])

superposed_B = magpy.getB(loop_1, observer_positions) + magpy.getB(loop_2, observer_positions)

collection_B = magpy.getB(loop_collection, observer_positions)

print("multiple observers at positions ")
print(observer_positions)

print("superposed_B =")
print(superposed_B)

print("collection_B =")
print(collection_B)

print("allclose =")
print(np.allclose(superposed_B, collection_B))

print("this fails... but is it a bug or a feature!?")

# my output:
##numpy version = 1.21.0
##magpylib version = 4.0.0rc1
##scipy version = 1.7.0
##Single observer at position = [0 0 0]
##superposed_B =
##[-0.00530893 -0.3464619  -0.24151183]
##collection_B =
##[-0.00530893 -0.3464619  -0.24151183]
##allclose =
##True
##Single observer at position = [1 1 1]
##superposed_B =
##[ 0.02117461 -0.42108487 -0.26771884]
##collection_B =
##[ 0.02117461 -0.42108487 -0.26771884]
##allclose =
##True
##multiple observer at positions 
##[[0 0 0]
## [1 1 1]]
##superposed_B =
##[[-0.00530893 -0.3464619  -0.24151183]
## [ 0.02117461 -0.42108487 -0.26771884]]
##collection_B =
##[[ 0.33011425 -0.1120442  -0.21807005]
## [ 0.02117461 -0.01336429 -0.00781032]]
##allclose =
##False
##this fails... but is it a bug or a feature!?

```

**Expected behavior**
Based on the docstring, I would have expected the collection B to be the same as the superposition of B. Perhaps there is something going on under the hood that I am missing, and this behavior is desired, but the docstring states that arrays of observer points of shape (n1, n2, ..., 3) are ok. Therefore, I would have expected the B fields at the observer positions in the last test to be equal. I also tested with higher dimensional arrays, (eg grid arrays (n1, n2, 3) similar to some of the getB examples in the doku), and get the same problem.

Thanks for the nice package, I'll be sure to cite you if we publish the simulations that I'm writing :)

**Comment by Alexboiboi** [comment]
Hi @apdioguardi, thank you for bringing this to our attention! 

This is definitely a bug !

Current work on a fix is at https://github.com/magpylib/magpylib/tree/getBH-pixel-tiling-bug and will be merged in a coming pull request. 

Since 4.0.0rc1 has not been released yet, may I ask you which branch did you use for this example? main?

I am also glad you discovered this before the official release of v4 🙏

**Comment by Alexboiboi** [comment]
Minimal example that shows the issue 
```python
import magpylib as magpy
import numpy as np

loop_1 = magpy.current.Loop(current=10000, diameter=20, position=(1, 20, 10))
loop_1.rotate_from_angax([45,90], "x")

loop_2 = magpy.current.Loop(current=2000, diameter=40, position=(20, 10, 1))
loop_2.rotate_from_angax([45,90], "y")

loop_collection = magpy.Collection(loop_1, loop_2)

observer_positions = [[0, 0, 0], [1, 1, 1]]

B1 = magpy.getB(loop_1, observer_positions)
B2 = magpy.getB(loop_2, observer_positions)
superposed_B = B1 + B2

collection_B = magpy.getB(loop_collection, observer_positions)

np.testing.assert_allclose(superposed_B, collection_B)
```

**Comment by apdioguardi** [comment]
> Hi @apdioguardi, thank you for bringing this to our attention!
> 
> This is definitely a bug !
> 
> Current work on a fix is at https://github.com/magpylib/magpylib/tree/getBH-pixel-tiling-bug and will be merged in a coming pull request.
> 
> Since 4.0.0rc1 has not been released yet, may I ask you which branch did you use for this example? main?
> 
> I am also glad you discovered this before the official release of v4 🙏

Oh sorry, yes I just installed directly from main via pip+git.

**Comment by Alexboiboi** [comment]
Hi @apdioguardi, we have just released a 4.0.0rc1 version which also includes the fix to the bug you discovered.

Feel free to give us more feedback on this pre-release, since the official release is now imminent ;)

Thanks

BTW you can install the latest pre-release with:
```bash
pip install magpylib --pre
```

**Comment by apdioguardi** [comment]
> Hi @apdioguardi, we have just released a 4.0.0rc1 version which also includes the fix to the bug you discovered.
> 
> Feel free to give us more feedback on this pre-release, since the official release is now imminent ;)
> 
> Thanks
> 
> BTW you can install the latest pre-release with:
> 
> ```shell
> pip install magpylib --pre
> ```

Thanks I just tested my more complex collection of current sources against my earlier version where I manually superposed the B-fields and everything looks good! I will be sure to let you know if there are any additional bugs.

I am starting to play with collections of magpylib.magnet.CylinderSegment today. So far I have only tested a single magnetized cylindrical shell against some previous calculations of B along the cylinder axis using an array of polarized dipole moments in the same configuration. This looked good to within the numerical precision of our old calculation.
